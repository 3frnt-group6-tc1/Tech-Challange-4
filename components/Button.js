import React from 'react';
import { TouchableOpacity, Text, StyleSheet, ActivityIndicator } from 'react-native';
import { useTheme } from '../contexts/ThemeContext';

export const Button = ({ 
  title, 
  onPress, 
  variant = 'primary', 
  size = 'md', 
  disabled = false, 
  loading = false,
  style,
  textStyle 
}) => {
  const { theme } = useTheme();
  
  const getButtonStyle = () => {
    const baseStyle = {
      borderRadius: theme.borderRadius.sm,
      alignItems: 'center',
      justifyContent: 'center',
      flexDirection: 'row',
    };

    const sizeStyles = {
      sm: { paddingVertical: theme.spacing.sm, paddingHorizontal: theme.spacing.md },
      md: { paddingVertical: theme.spacing.md, paddingHorizontal: theme.spacing.lg },
      lg: { paddingVertical: theme.spacing.lg, paddingHorizontal: theme.spacing.xl },
    };

    const variantStyles = {
      primary: {
        backgroundColor: theme.colors.primary,
      },
      secondary: {
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: theme.colors.primary,
      },
      danger: {
        backgroundColor: theme.colors.error,
      },
      success: {
        backgroundColor: theme.colors.success,
      },
    };

    return [
      baseStyle,
      sizeStyles[size],
      variantStyles[variant],
      disabled && { opacity: 0.6 },
      style,
    ];
  };

  const getTextStyle = () => {
    const baseTextStyle = {
      fontSize: theme.fontSize[size],
      fontWeight: 'bold',
    };

    const variantTextStyles = {
      primary: { color: '#FFFFFF' },
      secondary: { color: theme.colors.primary },
      danger: { color: '#FFFFFF' },
      success: { color: '#FFFFFF' },
    };

    return [
      baseTextStyle,
      variantTextStyles[variant],
      textStyle,
    ];
  };

  return (
    <TouchableOpacity
      style={getButtonStyle()}
      onPress={onPress}
      disabled={disabled || loading}
      activeOpacity={0.8}
    >
      {loading && (
        <ActivityIndicator 
          size="small" 
          color={variant === 'secondary' ? theme.colors.primary : '#FFFFFF'} 
          style={{ marginRight: theme.spacing.sm }}
        />
      )}
      <Text style={getTextStyle()}>{title}</Text>
    </TouchableOpacity>
  );
};
