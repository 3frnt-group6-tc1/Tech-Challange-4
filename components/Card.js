import React from "react";
import { View, StyleSheet } from "react-native";
import { useTheme } from "../contexts/ThemeContext";

export const Card = ({
  children,
  style,
  padding = "md",
  margin = "sm",
  ...otherProps
}) => {
  const { theme } = useTheme();

  const cardStyle = [
    styles.card,
    {
      backgroundColor: theme.colors.card,
      padding: theme.spacing[padding],
      margin: theme.spacing[margin],
      borderRadius: theme.borderRadius.md,
      shadowColor: theme.colors.shadow,
    },
    ...(Array.isArray(style) ? style : [style]),
  ].filter(Boolean);

  return (
    <View style={cardStyle} {...otherProps}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
});
