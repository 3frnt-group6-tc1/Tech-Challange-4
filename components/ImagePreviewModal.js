import {
  Modal,
  View,
  Image,
  TouchableOpacity,
  StyleSheet,
  Text,
} from "react-native";

const ImagePreviewModal = ({ visible, imageUri, onClose }) => {
  return (
    <Modal
      visible={visible}
      transparent
      animationType="fade"
      onRequestClose={onClose}
    >
      <View style={styles.overlay}>
        <TouchableOpacity style={styles.closeButton} onPress={onClose}>
          <Text>‚ùå</Text>
        </TouchableOpacity>
        <Image
          source={{ uri: imageUri }}
          style={styles.image}
          resizeMode="contain"
        />
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: "rgba(0,0,0,0.9)",
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "90%",
    height: "70%",
    borderRadius: 12,
  },
  closeButton: {
    position: "absolute",
    top: 30,
    right: 30,
    zIndex: 2,
  },
});

export default ImagePreviewModal;
