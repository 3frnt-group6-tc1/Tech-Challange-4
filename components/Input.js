import React from 'react';
import { TextInput, View, Text, StyleSheet } from 'react-native';
import { useTheme } from '../contexts/ThemeContext';

export const Input = ({ 
  label, 
  value, 
  onChangeText, 
  placeholder, 
  secureTextEntry = false,
  keyboardType = 'default',
  autoCapitalize = 'sentences',
  error,
  style,
  ...props 
}) => {
  const { theme } = useTheme();
  
  const inputStyle = [
    styles.input,
    {
      backgroundColor: theme.colors.surface,
      borderColor: error ? theme.colors.error : theme.colors.border,
      color: theme.colors.text,
      borderRadius: theme.borderRadius.sm,
      padding: theme.spacing.md,
      fontSize: theme.fontSize.md,
    },
    style,
  ];

  return (
    <View style={styles.container}>
      {label && (
        <Text style={[styles.label, { color: theme.colors.text, fontSize: theme.fontSize.sm }]}>
          {label}
        </Text>
      )}
      <TextInput
        style={inputStyle}
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        placeholderTextColor={theme.colors.textSecondary}
        secureTextEntry={secureTextEntry}
        keyboardType={keyboardType}
        autoCapitalize={autoCapitalize}
        {...props}
      />
      {error && (
        <Text style={[styles.errorText, { color: theme.colors.error }]}>
          {error}
        </Text>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 16,
  },
  label: {
    fontWeight: '600',
    marginBottom: 8,
  },
  input: {
    borderWidth: 1,
    minHeight: 48,
  },
  errorText: {
    fontSize: 12,
    marginTop: 4,
  },
});
