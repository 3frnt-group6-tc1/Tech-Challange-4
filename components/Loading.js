import React from 'react';
import { View, ActivityIndicator, Text, StyleSheet } from 'react-native';
import { useTheme } from '../contexts/ThemeContext';

export const Loading = ({ message = 'Carregando...' }) => {
  const { theme } = useTheme();

  return (
    <View style={[styles.container, { backgroundColor: theme.colors.background }]}>
      <ActivityIndicator 
        size="large" 
        color={theme.colors.primary} 
      />
      <Text style={[styles.message, { color: theme.colors.text }]}>
        {message}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  message: {
    marginTop: 16,
    fontSize: 16,
    textAlign: 'center',
  },
});