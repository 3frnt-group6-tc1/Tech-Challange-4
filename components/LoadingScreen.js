import React from 'react';
import { View, Text, ActivityIndicator, StyleSheet } from 'react-native';
import { useTheme } from '../contexts/ThemeContext';

const LoadingScreen = ({ message = 'Carregando...' }) => {
  const { theme } = useTheme();

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: theme.colors.background,
      padding: 20,
    },
    text: {
      marginTop: 20,
      fontSize: 16,
      color: theme.colors.text,
      textAlign: 'center',
    },
  });

  return (
    <View style={styles.container}>
      <ActivityIndicator size="large" color={theme.colors.primary} />
      <Text style={styles.text}>{message}</Text>
    </View>
  );
};

export default LoadingScreen;