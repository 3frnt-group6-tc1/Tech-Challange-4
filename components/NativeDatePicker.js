import React, { useState } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Modal,
  Platform,
} from 'react-native';
import { useTheme } from '../contexts/ThemeContext';

const NativeDatePicker = ({ 
  value, 
  onChange, 
  mode = 'datetime', 
  placeholder = 'Selecionar data',
  error = false 
}) => {
  const { theme } = useTheme();
  const [showModal, setShowModal] = useState(false);
  const [tempDate, setTempDate] = useState(value || new Date());

  const formatDate = (date) => {
    if (!date) return placeholder;
    
    try {
      if (mode === 'datetime') {
        return date.toLocaleString('pt-BR', {
          day: '2-digit',
          month: '2-digit',
          year: 'numeric',
          hour: '2-digit',
          minute: '2-digit',
        });
      } else if (mode === 'time') {
        return date.toLocaleTimeString('pt-BR', {
          hour: '2-digit',
          minute: '2-digit',
        });
      } else {
        return date.toLocaleDateString('pt-BR', {
          day: '2-digit',
          month: '2-digit',
          year: 'numeric',
        });
      }
    } catch (error) {
      return placeholder;
    }
  };

  const handleConfirm = () => {
    onChange(tempDate);
    setShowModal(false);
  };

  const handleCancel = () => {
    setTempDate(value || new Date());
    setShowModal(false);
  };

  const handlePress = () => {
    setTempDate(value || new Date());
    setShowModal(true);
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={[
          styles.pickerButton,
          {
            borderColor: error ? '#ff4444' : theme.colors.border,
            backgroundColor: theme.colors.surface,
          }
        ]}
        onPress={handlePress}
      >
        <Text style={[
          styles.pickerText,
          { 
            color: value ? theme.colors.text : theme.colors.textSecondary 
          }
        ]}>
          {formatDate(value)}
        </Text>
        <Text style={[styles.pickerIcon, { color: theme.colors.primary }]}>
          ðŸ“…
        </Text>
      </TouchableOpacity>

      <Modal
        visible={showModal}
        transparent
        animationType="slide"
        onRequestClose={handleCancel}
      >
        <View style={styles.modalOverlay}>
          <View style={[styles.modalContent, { backgroundColor: theme.colors.surface }]}>
            <Text style={[styles.modalTitle, { color: theme.colors.text }]}>
              Selecionar {mode === 'datetime' ? 'Data e HorÃ¡rio' : mode === 'time' ? 'HorÃ¡rio' : 'Data'}
            </Text>
            
            <View style={styles.dateInputs}>
              <View style={styles.dateRow}>
                <Text style={[styles.label, { color: theme.colors.text }]}>Dia:</Text>
                <TouchableOpacity
                  style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                  onPress={() => setTempDate(new Date(tempDate.getTime() - 24 * 60 * 60 * 1000))}
                >
                  <Text style={[styles.buttonText, { color: theme.colors.text }]}>-</Text>
                </TouchableOpacity>
                <Text style={[styles.dateValue, { color: theme.colors.text }]}>
                  {String(tempDate.getDate()).padStart(2, '0')}
                </Text>
                <TouchableOpacity
                  style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                  onPress={() => setTempDate(new Date(tempDate.getTime() + 24 * 60 * 60 * 1000))}
                >
                  <Text style={[styles.buttonText, { color: theme.colors.text }]}>+</Text>
                </TouchableOpacity>
              </View>

              <View style={styles.dateRow}>
                <Text style={[styles.label, { color: theme.colors.text }]}>MÃªs:</Text>
                <TouchableOpacity
                  style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                  onPress={() => setTempDate(new Date(tempDate.getFullYear(), tempDate.getMonth() - 1, tempDate.getDate()))}
                >
                  <Text style={[styles.buttonText, { color: theme.colors.text }]}>-</Text>
                </TouchableOpacity>
                <Text style={[styles.dateValue, { color: theme.colors.text }]}>
                  {String(tempDate.getMonth() + 1).padStart(2, '0')}
                </Text>
                <TouchableOpacity
                  style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                  onPress={() => setTempDate(new Date(tempDate.getFullYear(), tempDate.getMonth() + 1, tempDate.getDate()))}
                >
                  <Text style={[styles.buttonText, { color: theme.colors.text }]}>+</Text>
                </TouchableOpacity>
              </View>

              <View style={styles.dateRow}>
                <Text style={[styles.label, { color: theme.colors.text }]}>Ano:</Text>
                <TouchableOpacity
                  style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                  onPress={() => setTempDate(new Date(tempDate.getFullYear() - 1, tempDate.getMonth(), tempDate.getDate()))}
                >
                  <Text style={[styles.buttonText, { color: theme.colors.text }]}>-</Text>
                </TouchableOpacity>
                <Text style={[styles.dateValue, { color: theme.colors.text }]}>
                  {tempDate.getFullYear()}
                </Text>
                <TouchableOpacity
                  style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                  onPress={() => setTempDate(new Date(tempDate.getFullYear() + 1, tempDate.getMonth(), tempDate.getDate()))}
                >
                  <Text style={[styles.buttonText, { color: theme.colors.text }]}>+</Text>
                </TouchableOpacity>
              </View>

              {(mode === 'datetime' || mode === 'time') && (
                <>
                  <View style={styles.dateRow}>
                    <Text style={[styles.label, { color: theme.colors.text }]}>Hora:</Text>
                    <TouchableOpacity
                      style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                      onPress={() => setTempDate(new Date(tempDate.getTime() - 60 * 60 * 1000))}
                    >
                      <Text style={[styles.buttonText, { color: theme.colors.text }]}>-</Text>
                    </TouchableOpacity>
                    <Text style={[styles.dateValue, { color: theme.colors.text }]}>
                      {String(tempDate.getHours()).padStart(2, '0')}
                    </Text>
                    <TouchableOpacity
                      style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                      onPress={() => setTempDate(new Date(tempDate.getTime() + 60 * 60 * 1000))}
                    >
                      <Text style={[styles.buttonText, { color: theme.colors.text }]}>+</Text>
                    </TouchableOpacity>
                  </View>

                  <View style={styles.dateRow}>
                    <Text style={[styles.label, { color: theme.colors.text }]}>Minuto:</Text>
                    <TouchableOpacity
                      style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                      onPress={() => setTempDate(new Date(tempDate.getTime() - 60 * 1000))}
                    >
                      <Text style={[styles.buttonText, { color: theme.colors.text }]}>-</Text>
                    </TouchableOpacity>
                    <Text style={[styles.dateValue, { color: theme.colors.text }]}>
                      {String(tempDate.getMinutes()).padStart(2, '0')}
                    </Text>
                    <TouchableOpacity
                      style={[styles.numberButton, { backgroundColor: theme.colors.background }]}
                      onPress={() => setTempDate(new Date(tempDate.getTime() + 60 * 1000))}
                    >
                      <Text style={[styles.buttonText, { color: theme.colors.text }]}>+</Text>
                    </TouchableOpacity>
                  </View>
                </>
              )}
            </View>

            <View style={styles.buttonRow}>
              <TouchableOpacity
                style={[styles.actionButton, { backgroundColor: theme.colors.error }]}
                onPress={handleCancel}
              >
                <Text style={styles.actionButtonText}>Cancelar</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={[styles.actionButton, { backgroundColor: theme.colors.primary }]}
                onPress={handleConfirm}
              >
                <Text style={styles.actionButtonText}>Confirmar</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 16,
  },
  pickerButton: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 8,
    padding: 12,
    minHeight: 48,
  },
  pickerText: {
    fontSize: 16,
    flex: 1,
  },
  pickerIcon: {
    fontSize: 18,
    marginLeft: 8,
  },
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalContent: {
    width: '90%',
    maxWidth: 400,
    borderRadius: 12,
    padding: 20,
  },
  modalTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
    marginBottom: 20,
  },
  dateInputs: {
    marginBottom: 20,
  },
  dateRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 15,
  },
  label: {
    fontSize: 16,
    fontWeight: '600',
    width: 60,
  },
  numberButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  dateValue: {
    fontSize: 18,
    fontWeight: 'bold',
    minWidth: 60,
    textAlign: 'center',
  },
  buttonRow: {
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  actionButton: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderRadius: 8,
  },
  actionButtonText: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default NativeDatePicker;
