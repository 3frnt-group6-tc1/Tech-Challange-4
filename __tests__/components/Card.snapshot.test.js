import React from "react";
import { render } from "@testing-library/react-native";
import { Text } from "react-native";
import { Card } from "../../components/Card";

// Mock the useTheme hook with consistent values
jest.mock("../../contexts/ThemeContext", () => ({
  useTheme: () => ({
    theme: {
      colors: {
        card: "#FFFFFF",
        shadow: "rgba(0, 0, 0, 0.1)",
      },
      spacing: {
        xs: 4,
        sm: 8,
        md: 16,
        lg: 24,
        xl: 32,
      },
      borderRadius: {
        md: 12,
      },
    },
  }),
}));

describe("Card Component - Snapshot Tests", () => {
  it("should match snapshot with default props", () => {
    const tree = render(
      <Card>
        <Text>Default Card</Text>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with custom padding", () => {
    const tree = render(
      <Card padding="lg">
        <Text>Large Padding Card</Text>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with custom margin", () => {
    const tree = render(
      <Card margin="xl">
        <Text>Extra Large Margin Card</Text>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with custom styles", () => {
    const tree = render(
      <Card style={{ backgroundColor: "red", borderWidth: 2 }}>
        <Text>Custom Styled Card</Text>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with all props combined", () => {
    const tree = render(
      <Card
        padding="xs"
        margin="lg"
        style={{ opacity: 0.8, transform: [{ scale: 1.1 }] }}
      >
        <Text>All Props Card</Text>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with no children", () => {
    const tree = render(<Card />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with multiple children", () => {
    const tree = render(
      <Card>
        <Text>First Child</Text>
        <Text>Second Child</Text>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with nested cards", () => {
    const tree = render(
      <Card padding="lg">
        <Text>Outer Card</Text>
        <Card padding="sm" margin="xs">
          <Text>Inner Card</Text>
        </Card>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
