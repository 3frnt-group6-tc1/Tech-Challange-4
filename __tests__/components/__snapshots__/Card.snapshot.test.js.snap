// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Card Component - Snapshot Tests should match snapshot with all props combined 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 24,
        "padding": 4,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
      {
        "opacity": 0.8,
        "transform": [
          {
            "scale": 1.1,
          },
        ],
      },
    ]
  }
>
  <Text>
    All Props Card
  </Text>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with custom margin 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 32,
        "padding": 16,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
    ]
  }
>
  <Text>
    Extra Large Margin Card
  </Text>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with custom padding 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 8,
        "padding": 24,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
    ]
  }
>
  <Text>
    Large Padding Card
  </Text>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with custom styles 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 8,
        "padding": 16,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
      {
        "backgroundColor": "red",
        "borderWidth": 2,
      },
    ]
  }
>
  <Text>
    Custom Styled Card
  </Text>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with default props 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 8,
        "padding": 16,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
    ]
  }
>
  <Text>
    Default Card
  </Text>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with multiple children 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 8,
        "padding": 16,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
    ]
  }
>
  <Text>
    First Child
  </Text>
  <Text>
    Second Child
  </Text>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with nested cards 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 8,
        "padding": 24,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
    ]
  }
>
  <Text>
    Outer Card
  </Text>
  <View
    style={
      [
        {
          "elevation": 4,
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 8,
        },
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "margin": 4,
          "padding": 8,
          "shadowColor": "rgba(0, 0, 0, 0.1)",
        },
      ]
    }
  >
    <Text>
      Inner Card
    </Text>
  </View>
</View>
`;

exports[`Card Component - Snapshot Tests should match snapshot with no children 1`] = `
<View
  style={
    [
      {
        "elevation": 4,
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
      },
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "margin": 8,
        "padding": 16,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
      },
    ]
  }
/>
`;
